### DISCUSSION TITLE
Identifying Principal Procedure in Synthea output

### DISCUSSION THREAD

**Discussion by TravisHaussler** [synthetichealth/synthea#1349] [Q&A]
Hi all - we are trying to gather data from the Synthea CSV outputs to match this standard California submission formatting:

(https://hcai.ca.gov/wp-content/uploads/2022/12/IP-format-and-file-specs-jan-2023.pdf)

While not the only issue we are facing, one clear issue we have is figuring out what the "Principal Procedure (Code/Date)" is for a given encounter.  In the procedures.csv output we can often see multiple procedures for a given encounter_id, but we cannot see how to identify which might be the Principal (and therefore which of the rest are the "Other Procedures").

We are surely open to looking for data in places other than the CSV Synthea files as well if there is a more logical output place to look/gather from.

P.S. We are also only doing a very basic mapping from SNOMED (what Synthea uses) to ICD-10 (what California uses) coding and losing some robustness in the 1->many true mapping ICD uses.  Does anyone have experience teasing ICD-10 codes out of Synthea more naturally?

**Comment by jawalonoski** [comment]
Unfortunately, the principal procedure code is not identified.

The principal diagnosis is given as the encounter reason.

What we have seen in claims -- generally speaking -- is that a claim has a "header" that includes a principal diagnosis and all related diagnosis codes (typically ICD-10)... and each claim has 1-to-N "lines" where each line contains a single procedure (typically HCPCS or CPT).

The best you can probably do is to convert all the procedures in an encounter to ICD-10-PCS (we do not include this functionality in Synthea, but we use UMLS for this mapping, and you can too if you are in the United States). If there are no procedures (i.e., an encounter with no procedures), use a generic examination or care code, or a medication prescription/administration code (if any), or a vaccination code (if any). Drop codes that don't map.  With the list of mapped procedures, simply enumerate them in order and assign the first to Principal, and so on, up to 24 other procedures.

Not perfect, but it might get you close.

**Reply by rileeki** [reply]
@jawalonoski 

> we use UMLS for this mapping, and you can too if you are in the United States

Could you expand on this?  I do have a UMLS license, but I've only been able to find a SNOMED to ICD-10-CM mapping - nothing for ICD-10-PCS.

**Reply by jawalonoski** [reply]
You have to build it yourself using `MRCONSO.RRF`. Search for `ICD10PCS` and `SNOMEDCT_US` and match up the concept identifiers.
